src/parser/Parser.ts

```diff
@@ -1187,6 +1188,36 @@ export class Parser {
         return annotation;
     }
 
+    private ternaryExpression(test?: Expression): TernaryExpression {
+        this.warnIfNotBrighterScriptMode('ternary operator');
+        if (!test) {
+            test = this.expression();
+        }
+        const questionMarkToken = this.advance();
+
+
+        //we are a ternary
+        const consequent = this.expression();
+        if (!this.check(TokenKind.Colon)) {
+            //got here in error
+            this.diagnostics.push({
+                ...DiagnosticMessages.malformedTernaryExpression(),
+                range: test.range
+            });
+        }
+
+        const colonToken = this.tryConsume(DiagnosticMessages.malformedTernaryExpression(), TokenKind.Colon);
+        const alternate = this.expression();
+
+        if (!consequent || !alternate) {
+            this.diagnostics.push({
```

Comment:
Looks like a malformed ternary will result into something that we'll still be able to transpile - will it work?

Reply:
Probably not. However, I've tried very hard to still produce partial AST and as much AST recovery as possible, which helps with language server stuff. 

What should malformed transpiled ternary statement look like? Throw an error? Default to a no-op? Write an intentional syntax error so the code doesn't pass the on-device compile phase?

I've kinda just been letting the transpiler write whatever.